test/unified_malloc_framework/memoryPool.hpp

```diff
@@ -251,6 +258,26 @@ TEST_P(umfPoolTest, multiThreadedMallocFreeRandomSizes) {
     }
 }
 
+TEST_P(umfMemTest, outOfMem) {
+    static constexpr size_t allocSize = 16;
+    auto hPool = pool.get();
+
+    std::vector<void *> allocations;
+
+    while (true) {
+        allocations.emplace_back(umfPoolMalloc(hPool, allocSize));
+        if (umfPoolGetLastAllocationError(hPool) ==
```

Comment:
Actually, before checking the `umfPoolGetLastAllocationError` you should first check if `umfPoolMalloc` returned NULL. `umfPoolGetLastAllocationError` returns a valid error code ONLY when previous allocation failed. It can return anything if the allocation succeeded.

Reply:
Done.